csrc/tensor_metadata.cpp

```diff
@@ -315,18 +318,30 @@ inferAndValidateAllocationSizesAndStrides(
   allocation_sizes.reserve(alloc.size());
   allocation_strides.reserve(alloc.size());
   for (IterDomain* id : alloc | TensorDomain::kNoReductions) {
+    auto it = active_ids.find(id);
+    NVF_ERROR(it != active_ids.end(), "Allocation domain is not complete");
```

Comment:
This will fail when used with tests in `./bin/test_layout_op`, I will look into it, but not in this PR.